src/class-plugin-notice-controller.php

```diff
@@ -126,21 +138,19 @@ public function handle_notice_submission_ajax() {
 	}
 
 	/**
-	 * Handles the checkConsentHref request made by an <amp-consent> via AJAX.
+	 * Consent Check endpoint.
 	 *
-	 * @since 1.0.0
+	 * @param WP_REST_Request $request API request object.
+	 * @return WP_REST_Response|mixed API response or error.
 	 */
-	public function handle_amp_check_consent_href_ajax() {
-		$active = false;
-
+	public function api_check_consent( $request ) {
 		$payload = file_get_contents( 'php://input' ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
 		if ( ! empty( $payload ) ) {
 			$payload = json_decode( $payload, true );
 			if ( ! empty( $payload['consentInstanceId'] ) && Cookie_Notice_AMP_Markup::AMP_INSTANCE === $payload['consentInstanceId'] ) {
 				$active = $this->notice->is_active();
 			}
 		}
-
-		wp_send_json( [ 'promptIfUnknown' => $active ], 200 );
+		return \rest_ensure_response( [ 'promptIfUnknown' => $active ] );
```

Comment:
This needs a default value for `$active`. When `$payload` is empty or doesn't have the `consentInstanceId`, it will be undefined. Re-adding the removed `$active = false;` at the top of the method will solve the issue.